### DISCUSSION TITLE
مسدود کردن اسپیدتست در مرزبان (برای کاهش خطر فیلتر)

### DISCUSSION THREAD

**Discussion by DeAref** [Gozargah/Marzban#489] [Show and tell]
### مثل اینکه دولت ها میتونن از طریق شکل پکت های ارسالی و دریافتی تشخیص بدن که سروری که این اطلاعات رو ارسال میکنه یک vpn هست 
مخصوصا پکت هایی که از سمت سایت های مثل speedtest میاد چون همیشه شکلشون یکیه.

**برای مسدود کردن این سایت طبق مراحل زیر پیش برید**: 

یه پوشه تو مسیر مرزبان ایجاد کنید:
```
mkdir -p /var/lib/marzban/assets/
```

فایل dlc.dat و geoip.dat را از گیتهاب دانلود کنید:
```
wget -O /var/lib/marzban/assets/geosite.dat https://github.com/v2fly/domain-list-community/releases/latest/download/dlc.dat
```
```
wget -O /var/lib/marzban/assets/geoip.dat https://github.com/v2fly/geoip/releases/latest/download/geoip.dat
```

در پوشه مرزبان فایل env رو ادیت کنید:
```
nano /opt/marzban/.env
```
و خط زیر رو اضافه کنید:
```
XRAY_ASSETS_PATH = "/var/lib/marzban/assets/"
```

حالا برید تو پنل و علامت ⚙️ رو بزنید تا فایل جیسون Xray رو براتون بیاره
حالا آبجکت routing رو پیدا کنید و با جیسون زیر جایگذین کنید:
```
"routing": {
        "domainStrategy": "IPIfNonMatch",
        "rules": [
            {
                "type": "field",
                "outboundTag": "blackhole",
                "ip": [
                    "geoip:private"
                ]
            },
            {
                "type": "field",
                "port": 53,
                "network": "tcp,udp",
                "outboundTag": "DNS-Internal"
            },
            {
                "type": "field",
                "outboundTag": "blackhole",
                "protocol": [
                    "bittorrent"
                ]
            },
            {
                "outboundTag": "blackhole",
                "domain": [
                    "speedtest.net",
                    "speedtest.ir",
                    "speedcheck.ir",
                    "fast.com",
                    "speedcheck.org",
                    "speed.io"
                ],
                "type": "field"
            }
        ]
    }
```

هنوز از کانفیگ خارج نشید. آرایه outbound رو پیدا کنید
![image](https://github.com/Gozargah/Marzban/assets/95649368/056f1393-fe4b-43c8-bc6c-13fa9bd1b366)
و قسمت BLOCK رو به blackhole تغییر بدید و تغییرات رو save کنید.

و در آخر دوباره ssh بزنید به سرور و دستور `marzban restart` رو بزنید تا دوباره اجرا بشه.
حالا سایت speedtest دیگه بالا نمیاد، اگر بازم سایتی بود که خواستید اضافه کنید به لیست بلاک برگردید به فایل جیسون xray که در بالا گفتم و دامین اش رو اضافه کنید.

**Comment by rbsdotnet** [comment]
سپاس عالی بود 
برای من کار کرد 
❤️❤️❤️❤️❤️

**Comment by mahsamcse** [comment]
دوست عزیز
من که این قسمت routing رو ادیت زدم برای فعال کردن وارپ
حالا چطوری این کد شمارو هم بهش اضافه کنم ؟
ینی میشه هم اون کدهای وارپ باشه هم این کد شما ؟

**Reply by DeAref** [reply]
اره میشه

**Comment by nsa14** [comment]
سلام
برای بستن همین سایت ها در سرور نود جیکار باید کنیم؟
همین مارو روی نود‌انجام دادم نشد و کار نمیکنه

**Comment by gapochat** [comment]
> سلام برای بستن همین سایت ها در سرور نود جیکار باید کنیم؟ همین مارو روی نود‌انجام دادم نشد و کار نمیکنه

نود موقع ساخت مرزبانو کلون میکنه و همین قانون تو نود هم اعمال میشه...
